clients/src/main/java/org/apache/kafka/common/record/EosLogEntry.java

```diff
@@ -197,26 +208,29 @@ protected LogRecord makeNext() {
     }
 
     public void setOffset(long offset) {
-        buffer.putLong(OFFSET_OFFSET, offset);
+        buffer.putLong(BASE_OFFSET_OFFSET, offset);
     }
 
+    @Override
     public void setCreateTime(long timestamp) {
         long currentTimestamp = maxTimestamp();
         // We don't need to recompute crc if the timestamp is not updated.
         if (timestampType() == TimestampType.CREATE_TIME && currentTimestamp == timestamp)
             return;
 
         byte attributes = attributes();
-        buffer.put(ATTRIBUTES_OFFSET, TimestampType.CREATE_TIME.updateAttributes(attributes));
-        buffer.putLong(TIMESTAMP_OFFSET, timestamp);
+        buffer.putShort(ATTRIBUTES_OFFSET, TimestampType.CREATE_TIME.updateAttributes(attributes));
+        buffer.putLong(MAX_TIMESTAMP_OFFSET, timestamp);
         long crc = computeChecksum();
         Utils.writeUnsignedInt(buffer, CRC_OFFSET, crc);
     }
 
+    @Override
     public void setLogAppendTime(long timestamp) {
         byte attributes = attributes();
-        buffer.put(ATTRIBUTES_OFFSET, TimestampType.LOG_APPEND_TIME.updateAttributes(attributes));
-        buffer.putLong(TIMESTAMP_OFFSET, timestamp);
+        buffer.putShort(ATTRIBUTES_OFFSET, TimestampType.LOG_APPEND_TIME.updateAttributes(attributes));
+        buffer.putLong(BASE_TIMESTAMP_OFFSET, timestamp);
```

Comment:
Maybe add comment explaining why we need to set the base timestamp as well for `log append time` but not necessary for `create time`.